cmake/Toolchain-Ubuntu-mingw32.cmake

```diff
@@ -23,3 +23,6 @@ set(CMAKE_FIND_ROOT_PATH /usr/${TOOLCHAIN_PREFIX})
 set(CMAKE_FIND_ROOT_PATH_MODE_PROGRAM NEVER)
 set(CMAKE_FIND_ROOT_PATH_MODE_LIBRARY ONLY)
 set(CMAKE_FIND_ROOT_PATH_MODE_INCLUDE ONLY)
+
+# add static libgcc libraries so there are no additional dependencies
+set(CMAKE_SHARED_LINKER_FLAGS "-static-libgcc -static-libstdc++ -static")
```

Comment:
Oh, I should have mentioned, that I'm using crosscompiling in another use case. 
I'm only using the compiled library libcodec2 and not the entire application. Maybe this should be clarified in the readme for this part.
The GitHub action would also be only for codec2 and not freedv-gui.
Do you think, this could interfere with the freedv-gui build scripts?

Reply:
> Do you think, this could interfere with the freedv-gui build scripts?

I'm actually not sure but I tend to err on the side of caution here, especially since freedv-gui has worked without needing to do this. Is it possible to add these flags when building your codec2-using application? I would imagine it'd accomplish the same thing if you could.

Reply:
Yes, I'm thinking the same. For me this PR and thoughts are only if someone (like me) really needs to cross compile codec2 without using freedv-gui. 

It would be interesting for me to know, if an additional cmake file within the cmake folder - like I'm doing in this PR - is affecting the build steps of your build script, used in freedv-gui? So does the script or the compiler check for all *.cmake files within cmake/ folder? This would help me in understanding, which changes in the build process could affect freedv-gui.

Reply:
> Actually, we already collect dependencies as part of the Windows installer generation for freedv-gui. See https://github.com/drowe67/freedv-gui/blob/master/cmake/GetDependencies.cmake.in for more details. I would suggest _not_ statically liking these in and instead use WINEPATH for the GitHub action per the screenshot I posted earlier.

Yes I agree we shouldn't be forcing static linking of other libraries into `libcodec2.dll`.  Other end users may want to use `libcodec2.dll` in different ways.  So best to revert https://github.com/drowe67/codec2/pull/307/commits/0c19e875987f3e812edb22dfeacb2a3ca5bf8bdd and work out a way to run the GitHub action without static linking.

The GitHub action to test the windows build only needs to (1) build libcodec2.dll (2) run a simple test program using `wine` of a Windows test runner, e.g. like the [example above](https://github.com/drowe67/codec2/pull/307#issuecomment-1042548228) from @tmiw.

I agree @DJ2LS - this PR is just aiming at `codec2`, I don't think it will affect `freedv-gui` as it has a separate Windows build system.

Reply:
I had look at the wine path, @tmiw showing in his screenshot - Thanks, I didn't knew, there are dll-dependencies stored! Once again I learned something 😃 


Okay, so we are doing a step back to https://github.com/drowe67/codec2/commit/0c19e875987f3e812edb22dfeacb2a3ca5bf8bdd  and I will re-add `-DGENERATE_CODEBOOK` to the description.
Just to be sure, I understood it correctly: The added files [Toolchain-Ubuntu-mingw64.cmake](https://github.com/drowe67/codec2/pull/307/files#diff-84e95f72fc39813a2fce3820aade044b4c9f4204a92db86eafa33f1cb70262cf) and [Toolchain-Ubuntu-mingw32.cmake](https://github.com/drowe67/codec2/pull/307/files#diff-09e0152c2af9b1be6d85a5d5d314f7afc447dcfad358305093ec0d7839d64cf7) can stay?


However, I would suggest adding some additional information to the readme if its okay for you:
 - path to the *.dll files which are possibly needed for running libcodec2.dll `(/usr/lib/gcc/...)`
 - making an additional example with statically linking needed dependencies -  maybe just with a compiler flag for cli and not within [Toolchain-Ubuntu-mingw64.cmake](https://github.com/drowe67/codec2/pull/307/files#diff-84e95f72fc39813a2fce3820aade044b4c9f4204a92db86eafa33f1cb70262cf) and [Toolchain-Ubuntu-mingw32.cmake](https://github.com/drowe67/codec2/pull/307/files#diff-09e0152c2af9b1be6d85a5d5d314f7afc447dcfad358305093ec0d7839d64cf7)


Regarding to the Github action test I created, I'm not sure if its making sense. If the usage of wine would fit as well, then this could be possibly much less complex. The example I created would have much more overhead, I guess --> Crosscompiling on Ubuntu --> Saving as artifact --> Download artifact on windows runner --> Run test on windows

Reply:
> The added files [Toolchain-Ubuntu-mingw64.cmake](https://github.com/drowe67/codec2/pull/307/files#diff-84e95f72fc39813a2fce3820aade044b4c9f4204a92db86eafa33f1cb70262cf) and [Toolchain-Ubuntu-mingw32.cmake](https://github.com/drowe67/codec2/pull/307/files#diff-09e0152c2af9b1be6d85a5d5d314f7afc447dcfad358305093ec0d7839d64cf7) can stay?

Yes, minus the CMAKE_SHARED_LINKER_FLAGS line.

> path to the *.dll files which are possibly needed for running libcodec2.dll (/usr/lib/gcc/...)

That can be pretty distro specific so I'm not sure it's appropriate to have specific paths. A general explanation about libgcc should be fine, though.